dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/DefaultIconService.java

```diff
@@ -130,15 +149,10 @@ public boolean iconExists(String key) {
   public void addIcon(@Nonnull Icon icon)
       throws BadRequestException, NotFoundException, SQLException {
 
-    if (!icon.isCustom()) {
+    if (icon.getOrigin() == null && !icon.isCustom()) {
```

Comment:
It's not obvious what this `origin` field has to do with an icon being custom or not. This is only set for default icons right? Can this then be moved into the `isCustom`?